### PR TITLE
BUG: Ensure .astype doesn't use PandasArray

### PR DESCRIPTION
**Pull request by TomAugspurger**
On 0.24.0rc1, it's possible to end up with a PandasArray in internals.

```python
In [8]: ser = pd.Series([1, 2])

In [9]: ser.astype(ser.array.dtype)._data.blocks[0]
Out[9]: ExtensionBlock: 2 dtype: int64
```

### CODE REVIEW

File: pandas/tests/series/test_internals.py [hunk 1]
```diff
@@ -315,6 +315,11 @@ def test_constructor_no_pandas_array(self):
         tm.assert_series_equal(ser, result)
         assert isinstance(result._data.blocks[0], IntBlock)
 
+    def test_astype_no_pandas_dtype(self):
+        ser = pd.Series([1, 2], dtype="int64")
```
Comment (gfyoung) [inline pandas/tests/series/test_internals.py]: Reference PR number as a comment.

File: pandas/core/internals/blocks.py [hunk 2]
```diff
@@ -530,6 +530,10 @@ def f(m, v, i):
         return self.split_and_operate(None, f, False)
 
     def astype(self, dtype, copy=False, errors='raise', values=None, **kwargs):
+        dtype = pandas_dtype(dtype)
```
Comment (jreback) [inline pandas/core/internals/blocks.py]: i don’t want to put this clause where the existing is?

**Comment by TomAugspurger** [comment]
Subclasses overriding Block._astype get the sanitized version as well,
which is arguably what we want.

File: pandas/core/internals/blocks.py (hunk 2 above)
Comment (jreback) [reply pandas/core/internals/blocks.py]: the issue here is that we do all of the work in _astype, so now this is obscuring things

File: pandas/core/internals/blocks.py [hunk 3]
```diff
@@ -591,23 +591,16 @@ def _astype(self, dtype, copy=False, errors='raise', values=None,
...
+        if isinstance(dtype, PandasDtype):
+            dtype = dtype.numpy_dtype
+
         # astype processing
         if is_dtype_equal(self.dtype, dtype):
             if copy:
                 return self.copy()
             return self
 
-        klass = None
```
Comment (TomAugspurger) [inline pandas/core/internals/blocks.py]: I believe that this is unnecessary now... The `.astype` should get us an array of the correct type, and from there we'll get the correct block in `make_block`.

**Comment by TomAugspurger** [comment]
All green.

I'm a bit concerned that this is the wrong place to fix the issue. We could alternatively put this check in `make_block` so that if a PandasArray or PandasDtype is there, the the array / dtype is converted to a NumPy version before proceeding...

**Comment by TomAugspurger** [comment]
Trying to move it to `make_block`.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jreback** [comment]
needs a retest, can you merge master

**Comment by jreback** [comment]
merged master; lets see

**Comment by jreback** [comment]
thanks @TomAugspurger

**Pull request merged**